### PR TITLE
Rubocop to the rescue

### LINKED ISSUE #20: RUBOCOP DAY
# Description

- Add Rubocop-Rails
- Add Rubocop-Rspec
- Swap to use double quotes (scandalous, I know)

### PR DESCRIPTION
**Pull request by exegeteio**
Fixes #20?

Added rubocop-rails and rubocop-rspec to the Gemfile. Set defaults to double quotes, and ran rubocop --auto-correct.

All tests appear to pass in Docker Compose.

# Feedback Please on Unfixed Exceptions

I feel like this can be disabled on the Application Controller, given it is security related and most developers will understand that this does.

```
app/controllers/application_controller.rb:13:3: C: Rails/LexicallyScopedActionFilter: create, new, update, edit, destroy are not explicitly defined on the class.
  before_action :authenticate_user!, only: %i[create new update edit destroy]
```

I'm not certain what @ChaelCodes' intent with `event_attendees`.  If a profile is deleted, should it also delete the attendance record?  Seems like it would break things if the attendance record exists, but the profile no longer does.

```
app/models/profile.rb:8:3: C: Rails/HasManyOrHasOneDependent: Specify a :dependent option.
  has_many :event_attendees
```

I do not know enough about `bulk` changes to know if adding the `bulk: true` option would cause issues or not.

```
db/migrate/20210606154646_add_devise_to_users.rb:6:5: C: Rails/BulkChangeTable: You can combine alter queries using bulk: true options.
    change_table :users do |t|
```

### CODE REVIEW

**Comment by exegeteio** [comment]
I forgot to add that I did not copy ALL of the configuration customization from Hunters Keepers.  Only the ones which were effecting the existing codebase.

**Comment by ChaelCodes** [comment]
`app/controllers/application_controller.rb:13:3`

Can we disable it on the line easily?

`app/models/profile.rb:8:3: C: Rails/HasManyOrHasOneDependent: Specify a :dependent option.
  has_many :event_attendees`

I'm really glad you're catching this one. We want to delete all.

`db/migrate/20210606154646_add_devise_to_users`

This migration shipped already, so I don't think there's any value in adding `bulk: true`.

**Comment by ChaelCodes** [comment]
Would you mind squashing the history a bit?

**Comment by exegeteio** [comment]
Fixed / commented out the last three issues.

@ChaelCodes If you're good with the changes, I will squash.  I wanted to be verbose with the rspec changes to ensure I could back any out if I made a mistake.

File: spec/features/users/show_spec.rb [hunk 1]
```diff
@@ -1,39 +1,45 @@
 # frozen_string_literal: true
 
-require 'rails_helper'
+require "rails_helper"
 
-describe 'Users#show' do
+describe "Users" do
```
Comment (ChaelCodes) [inline spec/features/users/show_spec.rb:5]: I actually want the action name. It indicates which page the feature spec starts on.

Comment (ChaelCodes) [reply spec/features/users/show_spec.rb:5]: Did Rubocop complain about it? If so, we can drop it.

File: spec/features/users/show_spec.rb [hunk 2]
```diff
@@ -1,39 +1,45 @@
...
   let(:user_profile) do
     create(:user)
   end
 
-  before(:each) do
+  before do
```
Comment (ChaelCodes) [inline spec/features/users/show_spec.rb:10]: I need to fix this rule. You don't have to do it, but I prefer the explicit `before(:each)`.

File: spec/features/users/show_spec.rb [hunk 3]
```diff
@@ -1,39 +1,45 @@
...
 
-  context 'when no user logged in' do
+  context "when no user signed in" do
     let(:user) { nil }
 
-    it 'demands signup' do
-      expect(page).to have_content 'You need to sign in or sign up before continuing.'
-      expect(page).to_not have_content user_profile.name
+    it "demands signup" do
+      expect(page).to have_content "You need to sign in or sign up before continuing."
```
Comment (ChaelCodes) [inline spec/features/users/show_spec.rb:19]: Can we disable multiple expects for feature specs? I could find and fix this rule later, but these are expensive tests!

File: spec/requests/event_attendees_spec.rb [hunk 4]
```diff
@@ -14,74 +14,73 @@
...
+
   before do
     sign_in user if user
   end
 
-  describe 'GET /index' do
+  describe "GET /index" do
     subject(:get_index) { get event_attendees_url }
 
-    let!(:event_attendee) { create :event_attendee }
```
Comment (ChaelCodes) [inline spec/requests/event_attendees_spec.rb:28]: This let renders the table on the page.

**Review by ChaelCodes — COMMENTED** [review]
There were a few small things about the feature specs and a `let!` that got dropped from the index. Other than those, it looked like mostly spacing and quotes.

Thanks for all your help with this!

File: spec/features/users/show_spec.rb (hunk 1 above)
Comment (exegeteio) [reply spec/features/users/show_spec.rb:5]: It did:

```RSpec/DescribeClass: The first argument to describe should be the class or module being tested.```

File: spec/features/users/show_spec.rb (hunk 2 above)
Comment (exegeteio) [reply spec/features/users/show_spec.rb:10]: Looks like I can make this work with:

```
RSpec/HookArgument:
  EnforcedStyle: each
```

I'll drop it in.

File: spec/features/users/show_spec.rb (hunk 3 above)
Comment (exegeteio) [reply spec/features/users/show_spec.rb:19]: It appears so.  Dropping this in fixed that issue:

```
RSpec/MultipleExpectations:
  Exclude:
    - 'spec/features/**/*'
```

File: spec/requests/event_attendees_spec.rb (hunk 4 above)
Comment (exegeteio) [reply spec/requests/event_attendees_spec.rb:28]: But the tests still pass... Is that a smell?  I deleted several of the `let!`s?

**Comment by exegeteio** [comment]
So the squash turned out to be a bit of a nightmare.  Shouldn't merge mid-branch.  

Semi started over.  Three commits, and can be squashed easier now.

File: spec/requests/event_attendees_spec.rb (hunk 4 above)
Comment (exegeteio) [reply spec/requests/event_attendees_spec.rb:28]: I was able to resolve the Rubocop error by having the `expect` check the `response.body` for the `profile_id`.  This fixed the cop and still passed.  Does that make sense?

Applied the same logic in `events_spec` and `profiles_spec`.

File: spec/requests/event_attendees_spec.rb [hunk 5]
```diff
@@ -30,7 +30,7 @@
 
     it "renders a successful response" do
       get_index
-      expect(response).to be_successful
+      expect(response.body).to include(event_attendee.profile_id.to_s)
```
Comment (ChaelCodes) [inline spec/requests/event_attendees_spec.rb:33]: Fancy! A spec improvement! ✨

**Review by ChaelCodes — APPROVED** [review]
This looks amazing, and I know updating Rubocop rules for an entire repo is a huge lift.

Thank you for your PR!!

**Pull request merged**